chef/cookbooks/glance/recipes/ceph.rb

```diff
@@ -24,17 +27,19 @@
 else
   # If external Ceph cluster will be used,
   # we need install ceph client packages
-  if node[:platform] == "suse" && File.exists?("/etc/ceph/ceph.client.admin.keyring")
-    package "ceph-common"
-    package "python-ceph"
-  else 
+  if File.exists?(ceph_conf) && File.exists?(admin_keyring)
+    if node[:platform] == "suse"
+      package "ceph-common"
+      package "python-ceph"
+    end
+  else
     return
   end
 end
 
 # If ceph.conf and admin keyring will be available 
 # we have to check ceph cluster status
-check_ceph = Mixlib::ShellOut.new("ceph -s | grep -q -e 'HEALTH_[OK|WARN]'")
+check_ceph = Mixlib::ShellOut.new("ceph -k #{admin_keyring} -c #{ceph_conf} -s | grep -q -e 'HEALTH_[OK|WARN]'")
```

Comment:
> It seems that with full escaping we will have problem with pipe |. 

Yeah, if you pass the the command in as a list (which btw has not much to do with full escaping). ```Mixlib::ShellOut``` will not spawn a shell to execute the command. So stuff like pipes and $VARIABLE expansion will not work. 

> Do you have some idea how to fix it?

Instead of using ```| grep``` you could filter the stdout of the command in ruby. It should be available via ```check_ceph.stdout```. This seems cleaner to me anyway. Or maybe the exitcode of ```ceph -k '#{admin_keyring}' -c '#{ceph_conf}' -s``` already gives an indication if the cluster is healthy.